src/app/functions/server/getBreaches.ts

```diff
@@ -51,46 +51,53 @@ export async function getBreachIcons (breaches: Breach[]): Promise<LogoMap> {
     return new Map()
   }
   isFetchingIcons = true
-  const breachDomains = breaches
-    .map(breach => breach.Domain)
-    .filter(breachDomain => breachDomain.length > 0)
-  const logoFolder = pathResolve(dirname(fileURLToPath(import.meta.url)), '../../../../public/logo_cache/')
-  try {
-    await mkdir(logoFolder)
-  } catch {
-    // Do nothing; if the directory already exists, that's fine.
-  }
-  const existingLogos = await readdir(logoFolder)
-  // TODO: Batch to limit memory use?
-  const logoMapElems = await Promise.all(breachDomains.map(breachDomain => {
-    return new Promise((resolve, reject) => {
-      const logoFilename = breachDomain.toLowerCase() + '.ico'
-      const logoPath = pathResolve(logoFolder, logoFilename)
-      if (existingLogos.includes(logoFilename)) {
-        resolve([breachDomain, `/logo_cache/${logoFilename}`])
-        return
-      }
-      get(`https://icons.duckduckgo.com/ip3/${breachDomain}.ico`, (response) => {
-        if (response.statusCode !== 200) {
-          resolve(null)
+  async function fetchBreachIcons () {
```

Comment:
This function's contents is exactly the same as before, just indented. The part outside the braces is new.